pkg/backends/calico/routes.go

```diff
@@ -149,6 +150,26 @@ func (rg *routeGenerator) getServiceForEndpoints(ep *v1.Endpoints) (*v1.Service,
 
 // getEndpointsForService retrieves the corresponding ep for the given svc
 func (rg *routeGenerator) getEndpointsForService(svc *v1.Service) (*v1.Endpoints, string) {
+	// do nothing if the svc is not
+	if (svc.Spec.Type != v1.ServiceTypeClusterIP) &&
+		(svc.Spec.Type != v1.ServiceTypeNodePort) &&
+		(svc.Spec.Type != v1.ServiceTypeLoadBalancer) {
+		log.WithFields(log.Fields{
+			"svc":  svc.Name,
+			"type": svc.Spec.Type,
+		}).Debug("getEndpointForService: svc is not a valid type, passing...")
+		return nil, ""
+	}
+	// also do nothing if the clusterIP is empty or None
+	if len(svc.Spec.ClusterIP) > 0 &&
+		svc.Spec.ClusterIP != "None" {
+		log.WithFields(log.Fields{
+			"svc":       svc.Name,
+			"clusterIP": svc.Spec.ClusterIP,
+		}).Debug("getEndpointsForService: svc clusterIP is not valid, passing...")
```

Comment:
"valid" is the wrong word here - I'd say "not specified".

Reply:
I also think this logic (and the bit above) shouldn't be in this function - it belongs in "advertiseThisService".